commit 8e4d926fd0621f40b10c504ab2681416a61d9ad1
Author: Ravindu Deshan <droidcasts@protonmail.com>
Date:   Sat Mar 15 14:28:53 2025 +0530

    nuke out

diff --git a/build_5.15.sh b/build_5.15.sh
index 765c697..e230740 100755
--- a/build_5.15.sh
+++ b/build_5.15.sh
@@ -15,7 +15,7 @@ if [ ! -f ".requirements" ]; then
 fi
 
 # Create necessary directories
-mkdir -p "${RDIR}/out" "${RDIR}/build" "${HOME}/toolchains"
+mkdir -p "${RDIR}/build" "${HOME}/toolchains"
 
 #init neutron-clang
 if [ ! -d "${HOME}/toolchains/neutron-clang" ]; then
@@ -45,8 +45,6 @@ export BUILD_CC="${HOME}/toolchains/neutron-clang/bin/clang"
 
 # Build options for the kernel
 export BUILD_OPTIONS="
--C ${RDIR} \
-O=${RDIR}/out \
 -j$(nproc) \
 ARCH=arm64 \
 CROSS_COMPILE=${BUILD_CROSS_COMPILE} \
@@ -76,7 +74,7 @@ build_kernel(){
     make ${BUILD_OPTIONS} Image || exit 1
 
     # Copy the built kernel to the build directory
-    cp "${RDIR}/out/arch/arm64/boot/Image" "${RDIR}/build"
+    cp "${RDIR}/arch/arm64/boot/Image" "${RDIR}/build"
 
     echo -e "\n[INFO]: BUILD FINISHED..!"
 }
