commit c2b3ef99c7d99586238da10462f91150ea45f680
Author: Ravindu Deshan <Droidcasts@protonmail.com>
Date:   Fri Jan 17 15:00:05 2025 +0000

    Removed: -dirty string, commit hash and append localversion instead

diff --git a/arch/arm64/configs/custom.config b/arch/arm64/configs/custom.config
index 74e44aa2f..b022fb74d 100644
--- a/arch/arm64/configs/custom.config
+++ b/arch/arm64/configs/custom.config
@@ -45,3 +45,6 @@ CONFIG_KSU_SUSFS_HIDE_KSU_SUSFS_SYMBOLS=y
 CONFIG_KSU_SUSFS_SPOOF_CMDLINE_OR_BOOTCONFIG=y
 CONFIG_KSU_SUSFS_OPEN_REDIRECT=y
 CONFIG_KSU_SUSFS_SUS_SU=n
+
+#localversion
+CONFIG_LOCALVERSION_AUTO=n
diff --git a/scripts/setlocalversion b/scripts/setlocalversion
index c2e8455b3..dd32f379a 100755
--- a/scripts/setlocalversion
+++ b/scripts/setlocalversion
@@ -108,19 +108,6 @@ scm_version()
 			printf -- '-svn%s' "`git svn find-rev $head`"
 		fi
 
-		# Check for uncommitted changes.
-		# First, with git-status, but --no-optional-locks is only
-		# supported in git >= 2.14, so fall back to git-diff-index if
-		# it fails. Note that git-diff-index does not refresh the
-		# index, so it may give misleading results. See
-		# git-update-index(1), git-diff-index(1), and git-status(1).
-		if {
-			git --no-optional-locks status -uno --porcelain 2>/dev/null ||
-			git diff-index --name-only HEAD
-		} | grep -qvE '^(.. )?scripts/package'; then
-			printf '%s' -dirty
-		fi
-
 		# All done with git
 		return
 	fi
@@ -139,12 +126,6 @@ scm_version()
 			fi
 		fi
 
-		# Are there uncommitted changes?
-		# These are represented by + after the changeset id.
-		case "$hgid" in
-			*+|*+\ *) printf '%s' -dirty ;;
-		esac
-
 		# All done with mercurial
 		return
 	fi
@@ -200,19 +181,4 @@ fi
 # CONFIG_LOCALVERSION and LOCALVERSION (if set)
 res="${res}${CONFIG_LOCALVERSION}${LOCALVERSION}"
 
-# scm version string if not at a tagged commit
-if test "$CONFIG_LOCALVERSION_AUTO" = "y"; then
-	# full scm version string
-	res="$res$(scm_version)"
-else
-	# append a plus sign if the repository is not in a clean
-	# annotated or signed tagged state (as git describe only
-	# looks at signed or annotated tags - git tag -a/-s) and
-	# LOCALVERSION= is not specified
-	if test "${LOCALVERSION+set}" != "set"; then
-		scm=$(scm_version --short)
-		res="$res${scm:++}"
-	fi
-fi
-
 echo "$res"
