commit 75a7c07c13868f051ee1501347fb220c9aa0ef95
Author: @ravindu644 <droidcasts@protonmail.com>
Date:   Sun Jul 21 14:20:38 2024 +0530

    fatal error: 'linux/state_notifier.h' file not found

diff --git a/include/linux/state_notifier.h b/include/linux/state_notifier.h
new file mode 100644
index 000000000..6965cd189
--- /dev/null
+++ b/include/linux/state_notifier.h
@@ -0,0 +1,21 @@
+#ifndef __LINUX_STATE_NOTIFIER_H
+#define __LINUX_STATE_NOTIFIER_H
+
+#include <linux/notifier.h>
+
+#define STATE_NOTIFIER_ACTIVE		0x01
+#define STATE_NOTIFIER_SUSPEND		0x02
+#define STATE_NOTIFIER			"state_notifier"
+
+struct state_event {
+	void *data;
+};
+
+extern bool state_suspended;
+extern void state_suspend(void);
+extern void state_resume(void);
+int state_register_client(struct notifier_block *nb);
+int state_unregister_client(struct notifier_block *nb);
+int state_notifier_call_chain(unsigned long val, void *v);
+
+#endif /* _LINUX_STATE_NOTIFIER_H */
