commit 896574bace78ed509d9b7270c55a7c06c6f1e975
Author: Ravindu Deshan <Droidcasts@protonmail.com>
Date:   Wed Aug 7 07:04:19 2024 +0000

    fixed : multiple definition of 'is_space'

diff --git a/drivers/input/touchscreen/gt1151q_1695/gt1x_test.c b/drivers/input/touchscreen/gt1151q_1695/gt1x_test.c
index 6c18ab543e4f..e8b5762bfc91 100755
--- a/drivers/input/touchscreen/gt1151q_1695/gt1x_test.c
+++ b/drivers/input/touchscreen/gt1151q_1695/gt1x_test.c
@@ -243,17 +243,6 @@ int is_digit(char ch)
 		return 0;
 }
 
-int is_space(char ch)
-{
-	if (ch == ' ')
-
-		return 1;
-
-	else
-
-		return 0;
-}
-
 /* return value = float value * 1000; */
 long atof(char const *s)
 {
